Add best practices for managing content (#3004)
theforeman/foreman-documentation: guides/common/modules/ref_best-practices-for-lifecycle-environments.adoc

[id="best-practices-for-lifecycle-environments_{context}"]
= Best practices for lifecycle environments

* Use multiple lifecycle environment paths to implement multiple sequential stages of content consumption.
Each stage contains a defined set of content, for example in the _Production_ lifecycle environment.
* Automate the creation of lifecycle environments by using a Hammer script or an {ansible-docs-url}[Ansible playbook].

> * Default use case: Fixed stages in each lifecycle environment paths, for example _Development_, _Test_, and _Production_.

Reviewer: I think I see what you are doing here. This looks like a concept module. :-) 

I guess we can leave it in.

** Promote content views to lifecycle environments, for example from _Test_ to _Production_.
All content hosts consuming this content view or composite content view are able to install packages from the _Production_ lifecycle environment.
Note that these packages are not installed or updated automatically.
** If you encounter errors during patching content hosts, attach the host to a previous version of the content view.
This only affects the availability of packages but does not downgrade installed packages.

> * Alternative use case: Using stages in lifecycle environments for fixed content, for example quarterly updates, and only ever publish new minor versions when adding incremental updates from errata.

Reviewer: ```suggestion
* Alternative use case: Using stages in lifecycle environments for fixed content, for example, quarterly updates, and only publishing new minor versions with incremental updates from errata.
```

[id="best-practices-for-lifecycle-environments_{context}"]
= Best practices for lifecycle environments

* Use multiple lifecycle environment paths to implement multiple sequential stages of content consumption.
Each stage contains a defined set of content, for example in the _Production_ lifecycle environment.

> * Default use case: Fixed stages in each lifecycle environment paths, for example _Development_, _Test_, and _Production_.

Author: Applied.

** Promote content views to lifecycle environments.

> ** To make content from _Test_ available in _Production_, promote the content view version from _Test_ to _Production_.

Reviewer: These look like procedure steps. They break up the flow. Consider moving to another module, even if it is a high-level procedure.

Author: I have reworded some bullet points to make it sound less like instructions. What do you think?
